CMakeLists.txt

```diff
@@ -311,6 +311,9 @@ cmake_dependent_option(
     USE_C10D_GLOO "USE C10D GLOO" ON "USE_DISTRIBUTED;USE_GLOO" OFF)
 cmake_dependent_option(
     USE_C10D_NCCL "USE C10D NCCL" ON "USE_DISTRIBUTED;USE_NCCL" OFF)
+cmake_dependent_option(
+    USE_NCCL_WITH_UCC "Enable UCC support for ProcessGroupNCCL. Only available if USE_C10D_NCCL is on." ON
```

Comment:
Do we really need this cmake flag if we are anyways only conditionally loading the ucc lib at runtime? What is the downside if we don't have this flag at all and default to true? Would there be some issues in the case where ucc lib is not found?

Reply:
I added this based on https://github.com/pytorch/pytorch/pull/69552#issuecomment-990073975. Right now, it might seem a bit redundant, but in the future, when `ProcessGroupNCCL` starts to automatically search and load `torch_ucc.so` from the user's system, having a cmake flag will be helpful. In my opinion, if we are sure we will need one in the future, it won't be bad to have it now, but I can remove this flag if you want.